packages/integrations/node/src/serve-static.ts

```diff
@@ -0,0 +1,79 @@
+import path from "node:path";
+import url from "node:url";
+import send from "send";
+import type { IncomingMessage, ServerResponse } from "node:http";
+import type { Options } from "./types.js";
+import type { NodeApp } from "astro/app/node";
+
+/**
+ * Intended to be used only in the standalone mode.
+ * This is queried first, and if a static file is not found, it relegates to the SSR handler.
+ */
+export default function createListener(app: NodeApp, options: Options) {
+	const client = resolveClientDir(options);
+    return (req: IncomingMessage, res: ServerResponse, ssr: () => unknown) => {
+		if (req.url) {
+			let pathname = app.removeBase(req.url);
+			pathname = decodeURI(new URL(pathname, 'http://host').pathname);
+
+			const stream = send(req, pathname, {
+				root: client,
+				dotfiles: pathname.startsWith('/.well-known/') ? 'allow' : 'deny',
+			});
+
+			let forwardError = false;
+
+			stream.on('error', (err) => {
+				if (forwardError) {
+					console.error(err.toString());
+					res.writeHead(500);
+					res.end('Internal server error');
+					return;
+				}
+				// File not found, forward to the SSR handler
+				ssr();
+			});
+			stream.on('headers', (_res: ServerResponse) => {
+                // assets in dist/_astro are hashed and should get the immutable header
+				if (pathname.startsWith(`/${options.assets}/`)) {
+					// Taken from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#immutable
```

Comment:
The link doesn't explain the `max-age`, so we should add a comment that explains why we decided this number. Do you know why?

Reply:
This is the "far future" cache header, used for static whose name/path includes their digest hash. 1 year (31,536,000 seconds) is convention.